include/fmt/chrono.h

```diff
@@ -311,12 +319,30 @@ inline auto do_write(const std::tm& time, const std::locale& loc, char format,
     FMT_THROW(format_error("failed to format time"));
   str.clear();
   for (code_unit* p = buf; p != to_next; ++p) {
-    auto c = *p;
+    uint32_t c = static_cast<uint32_t>(*p);
+    if (sizeof(code_unit) == 2 && c >= 0xd800 && c <= 0xdfff) {
+      // surrogate pair
+      ++p;
+      if (p == to_next || (c & 0xfffffc00) != 0xd800 ||
```

Comment:
`0xfffffc00` -> `0xfc00` since code unit only has 2 bytes.

Reply:
Code using value ```0xfffffc00``` will work always if condition ```sizeof (code_unit) == 2``` is accidentally removed.
I specially made 4-byte masks so that the code always works. And because type of ```c``` is a ```uint32_t```.

If you think it is better to make a change, I will make it. Please write which option to choose.

Reply:
I suggest using 2-byte masks because surrogate pairs only make sense for UTF-16.